lib/compat/wordpress-6.2/default-filters.php

```diff
@@ -25,3 +25,10 @@
  */
 add_action( 'switch_theme', 'wp_theme_clean_theme_json_cached_data' );
 add_action( 'start_previewing_theme', 'wp_theme_clean_theme_json_cached_data' );
+add_action( 'save_post_wp_global_styles', array( 'WP_Theme_JSON_Resolver', 'clean_cached_data' ) );
```

Comment:
Nice. While this is not part of this PR, it helps to fix the issue, so we should do it. We'd need to also clean this data upon the following missing events: `activated_plugin`, `deactivated_plugin`.

Adding it for `upgrader_process_complete` is a bit more involved, it can be done in a follow-up separate PR.

Reply:
> We'd need to also clean this data upon the following missing events: activated_plugin, deactivated_plugin.

Done in https://github.com/WordPress/gutenberg/pull/45679/commits/8e2ffb8948a90eeb68272e202da6a7c7c97135a6

> We also need it for upgrader_process_complete.

Done in https://github.com/WordPress/gutenberg/pull/45679/commits/18f144346bc4bd2b2e4a3783a6a394a6d46d9fde